### ISSUE TITLE
VideoPress: allow users to upload videos from Gutenberg [2]

### ISSUE THREAD

**Issue by mmtr** [Automattic/jetpack#11194]
#### Steps to reproduce the issue

0. Start from a Jetpack site with an active Professional upgrade.
1. Go to Jetpack > Settings, and enable the VideoPress feature.
2. Go to Post > Add new.
3. Insert a video block.
4. Upload a video using the "Upload" button.
5. Save / publish / discard the post.
6. Go to Media > Library and check the video previously uploaded.

#### What I expected
The video is processed by VideoPress.
<img width="784" alt="screen shot 2019-01-24 at 15 42 33" src="https://user-images.githubusercontent.com/1233880/51685536-cfc54400-1fee-11e9-964a-8ff78fed41eb.png">

#### What happened instead
The video has not been processed by VideoPress.
<img width="781" alt="screen shot 2019-01-24 at 15 42 13" src="https://user-images.githubusercontent.com/1233880/51685497-b7edc000-1fee-11e9-97d9-f78061f1ef40.png">

**Comment by mmtr** [comment]
Relevant discussion in p9dueE-xm-p2.

**Uploading videos with the Media Library**
Note that the issue is specific to the “Upload” button. Videos uploaded through “Media Library” are transcoded without any problem.

This is a good indicator of what is going on. So let’s start analyzing how Jetpack uploads videos when using the Media Library.

The VideoPress module enqueues some scripts when it detects we are in a page where we can upload files ([#](https://github.com/Automattic/jetpack/blob/cdfe559613c989875050642189664f2cdafbd651/modules/videopress/class.jetpack-videopress.php#L128-L164)). These scripts changes how plupload creates new media items (plupload is the file uploader library used in the media modal).

Specifically, it makes an extra AJAX call when we select a video ([#](https://github.com/Automattic/jetpack/blob/cdfe559613c989875050642189664f2cdafbd651/modules/videopress/js/videopress-uploader.js#L121)) for retrieving a token and a /rest/v1.1/sites/:site/media/new URL ([#](https://github.com/Automattic/jetpack/blob/cdfe559613c989875050642189664f2cdafbd651/modules/videopress/class.videopress-ajax.php#L46-L72)) that are used as an authorization header and as the new upload URL ([#](https://github.com/Automattic/jetpack/blob/cdfe559613c989875050642189664f2cdafbd651/modules/videopress/js/videopress-plupload.js#L341-L353)).

In other words, videos are uploaded by making a request to https://public-api.wordpress.com/rest/v1.1/sites/:site/media/new that is authenticated with a X_UPLOAD_TOKEN token.

That means that we’re actually creating the new media items in WordPress.com.

**The real issue: /wp/v2/media endpoints**
Going back to original issue, what happens when we select a file using the “Upload” button? Gutenberg creates new media items using the /wp/v2/media endpoint.

Since the request is to the self-hosted site (https://example.com/wp/v2/media) and we don’t have in place any logic for handling these uploads, the video is not transcoded.

**Comment by mzakariya** [comment]
I've updated this support doc to **remove references** to the Video block, until this is addressed:

- https://jetpack.com/support/video-hosting/

I've updated this support doc to **note that you cannot upload via the Video block** until this is addressed:

- https://jetpack.com/support/video-block/

We should revert these changes to our user-facing documentation once this issue has been addressed.

**Comment by tmmbecker** [comment]
confusion about this reported in 2084562-zen

**Comment by kwight** [comment]
Likely related to https://github.com/Automattic/jetpack/issues/7073.
